### ISSUE TITLE
rfc: csv database backups?

### ISSUE THREAD

**Issue by NoSharp** [NoSharp/portsoc-role-bot#1]
Hello,
Quite simply, should we store prior versions of the CSV database?
I think it'd be a good idea to store the past x versions of the database allowing a user to roll-back and roll-forward on database versions.

**Comment by jeetelongname** [comment]
It makes sense, In a sense making changes transactional would be a pretty safe and effective way of making sure changes are correct

**Comment by JavaRip** [comment]
I feel like this makes some assumptions about both our design and what the user wants.

- Are we going to have storage? What kind (filesystem / databse / something else)? 
- How are we going to protect the data we're storing (a user account system)? 
- Would it be possible to design a stateless solution where you upload a csv which specifies roles to be assigned, the roles are assigned and nothing is saved? (imo stateless seems preferable)

Have we confirmed a CSV is what users want? Probably, but we should confirm that before getting too attached to the idea - or we should confirm this is an assumption we're moving forward with

**Comment by NoSharp** [comment]
> Have we confirmed a CSV is what users want? Probably, but we should confirm that before getting too attached to the idea - or we 
> should confirm this is an assumption we're moving forward with

I spoke to @manighahrmani about this and was provided with sample data I'll share with you on discord.

**Comment by manighahrmani** [comment]
The CSV may only contain up number and course at the start. It is going to be a bit tricky ethics wise dealing with actual names. So let it be a matter of assigning roles based on up numbers and formatting nicknames with the name provided by the user.

**Comment by NoSharp** [comment]
> The CSV may only contain up number and course at the start. It is going to be a bit tricky ethics wise dealing with actual names. So let it be a matter of assigning roles based on up numbers and formatting nicknames with the name provided by the user.

Yes, this is what I was planning on, as the pre-processor will just re-generate a new CSV file with certain unnessecary fields redacted.
We'll need to decide on what data storage medium we're able to run and what would work well.
I personally think because the CSV file is probably ~300 entries to just to generate a new CSV file. 
We could always look at parsing the CSV then creating a schema, however I personally think for the time being it's over-complicating it to say the least at the moment.
